### PR TITLE
Update/Close Share Modal When Clicked Outside

### PR DESCRIPTION
**Pull request by TheSteveK**
Per c/jzUXOAT8-tr This pr will cause the Share modal to close when a user clicks outside the modal (on to the project editor screen behind the modal).

Big thanks to @jcheringer for the assist.  We discussed whether this would better work as an inline function on the modal wrapper, or by the current method of building the modal as part of the function that determines whether or not the user clicks on the project or the modal.  (Bubbling is hard, y'all!)

I'm not sure if there's a more react-centric way of doing this, so I'd love to hear opinions :)

Testing instructions:

Checkout and run the PR.
Create or edit a Project (the project will need to be published)
Click on Sharing within the editor tab to see the modal.
Click outside the modal to see if it closes
Make sure that clicking the modal itself does not close the modal.

### CODE REVIEW

**Review by jcheringer — APPROVED** [review]
Well, looks good to me but, since we worked together on this one, let's wait for one more approval. 😃

**Review by ice9js — APPROVED** [review]
LGTM!

**Pull request merged**
